app/helpers/payments_helper.rb

```diff
@@ -4,32 +4,63 @@ module PaymentsHelper
-  # must respond to :membership_expire_date; else must respond to :branding_expire_date
+  # must respond to :entity_expire_date
   #
   # @return [String] - the HTML <span> string
   def expire_date_label_and_value(entity)
-    if entity.is_a? User
-      expire_date = entity.membership_expire_date
-      expire_after_tooltip_title = "#{t('users.show.membership_expire_date_tooltip')}"
+
+    expire_date = entity_expire_date(entity)
+    t_scope = entity.is_a?(User) ? 'users' : 'companies' # TODO - should use polymorphism to handle this
+
+    expire_after_tooltip_title = t("#{t_scope}.show.term_expire_date_tooltip")
+    expire_label = t("#{t_scope}.show.term_paid_through")
+
+    if expire_date
+      tag.div do
+        concat tag.span "#{expire_label}: ", class: 'standard-label'
+        concat tag.span "#{expire_date}", class: payment_should_be_made_class(entity)
+        concat ' '
+        concat fas_tooltip(expire_after_tooltip_title)
+      end
     else
-      expire_date = entity.branding_expire_date # Company
-      expire_after_tooltip_title = "#{t('companies.show.branding_fee_expire_date_tooltip')}"
+      field_or_none(expire_label, t('none_t'), label_class: 'standard-label')
     end
+  end
 
-    unless expire_date
-      return field_or_none("#{t('activerecord.attributes.payment.expire_date')}",
-                           "#{t('none_t')}", label_class: 'standard-label')
+
+  # @param entity [User or Company] - the entity with a possible membership expiration date
+  # @return [nil | date] - return nil if there is no expiration date (e.g. not a member), else
+  # the Date that the current membership term expires
+  # TODO - this is a smell (checking the class): polymorphism should be used instead. Default implementation should be in a parent class for User and Company
+  def entity_expire_date(entity = nil)
+    if entity && (entity.is_a?(User) || entity.is_a?(Company))
+      entity.is_a?(User) ? entity.membership_expire_date : entity.branding_expire_date
+    else
+      nil
     end
+  end
 
-    tag.p do
-      concat tag.span "#{t('activerecord.attributes.payment.expire_date')}: ",
-                      class: 'standard-label'
-      concat tag.span "#{expire_date}", class: expire_date_css_class(expire_date)
-      concat ' '
-      concat fas_tooltip(expire_after_tooltip_title)
+
+  # @return [String] - the scope (= key) to use when looking up I18n translations
+  # for this entity.  Default = 'users'
+  # TODO - this is a smell (checking the class): polymorphism should be used instead. Default implementation should be in a parent class for User and Company
+  def entity_i18n_scope(entity = nil)
+    (entity && entity.is_a?(Company)) ? 'companies' : 'users'
+  end
+
+
+  def payment_should_be_made_class(entity)
+    # TODO use the classes defined in ApplicationHelper
+    if entity.term_expired?
+      'No'
+    elsif entity.too_early_to_pay?
+      'Yes'
+    else
+      'Maybe'
```

Comment:
I know this is just the CSS class that we use to indicate whether the user should (or _maybe_ should) pay a fee or not.  However, the method name is confusing when coupled with the return values.

Perhaps a name like `payment_status_ok_indicator_css_class` would be better(?).  And, yes, maybe too wordy.

Reply:
excellent catch.  Made me realize it would be helpful to always make sure it's clear when we're dealing with _CSS classes_ vs. "normal" Ruby classes. 
I'll hit up the thesaurus to see if I can come up with something more succinct.  (Which is not my strong suit ;-) )
This also prompted me to enter a small chore story in PT to do some refactoring so we can simplify and standardize the 'yes/no/maybe' CSS classes and usage.